CDCgov/prime-reportstream
# Updated local Postgres to v14

This PR updates the local PostgreSQL database from v11 to v14.  Note you must delete the database volume for this update to take effect, which also means THAT ALL YOUR DB DATA WILL BE LOST UNLESS YOU BACK IT UP.

Test Steps:
1. Check out this branch.
2. Run the following commands to update the database.
```
docker compose -f ./docker-compose.postgres.yml down
docker volume rm prime-router_vol_postgresql_data
docker compose -f ./docker-compose.postgres.yml up -d
./gradlew resetDB
```
4. Start ReportStream and run the following commands to reload the DB
```
./gradlew reloadTables
./gradlew reloadSettings
```

## Changes
- Updated PostgreSQL to v14 locally

## Checklist

### Testing

### Process

## Linked Issues
- Fixes #issue

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #issue

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?
